olive/cli/auto_opt.py

```diff
@@ -0,0 +1,255 @@
+            choices=["fp16", "fp32", "int4", "int8"],
+            help=(
+                "The output precision of the optimized model. If not specified, "
+                "the default precision is fp32 for cpu and fp16 for gpu"
+            ),
+        )
+        auto_opt_config_group.add_argument(
+            "--excluded_passes",
+            type=str,
+            nargs="*",
+            help=(
+                "List of passes to disable for optimization, if not specified, "
+                "auto-opt will disable ModelBuilder/OrtPerfTuning by default."
+            ),
+        )
+        auto_opt_config_group.add_argument(
+            "--use_model_builder",
+            action="store_true",
+            help=(
+                "Whether to use model builder pass for optimization, enable only "
+                "when the model is supported by model builder"
+            ),
+        )
+
+        search_strategy_group = sub_parser.add_argument_group("search strategy options")
+        search_strategy_group.add_argument(
+            "--num_samples", type=int, default=5, help="Number of samples for search algorithm"
+        )
+        search_strategy_group.add_argument("--seed", type=int, default=0, help="Random seed for search algorithm")
+        search_strategy_group.add_argument(
+            "--search_order",
+            type=str,
+            default="joint",
+            choices=["joint", "pass-by-pass"],
+            help="Execution order for search strategy",
+        )
+        search_strategy_group.add_argument(
+            "--search_algorithm",
+            type=str,
+            default="tpe",
+            choices=["exhaustive", "tpe", "random"],
+            help="Search algorithm for search strategy",
+        )
+
+        # output options
+        output_group = sub_parser.add_argument_group("output options")
+        output_group.add_argument(
+            "--tempdir", default=None, type=str, help="Root directory for tempfile directories and files"
+        )
+        output_group.add_argument("-o", "--output_path", type=str, default="auto-opt-output", help="Output path")
+        # remote options
+        add_remote_options(sub_parser)
+        sub_parser.set_defaults(func=AutoOptCommand)
+
+    def _get_data_config(self) -> Dict:
+        with open(self.args.data_config_path) as f:
+            data_config = json.load(f)
+            data_config["name"] = "data_config"
+            return data_config
+
+    def run(self):
+        from olive.workflows import run as olive_run
+
+        set_tempdir(self.args.output_path)
+        with tempfile.TemporaryDirectory() as tempdir:
+            run_config = self.get_run_config(tempdir)
+            olive_run(run_config)
+            if is_remote_run(self.args):
+                # TODO(jambayk): point user to datastore with outputs or download outputs
+                # both are not implemented yet
+                return
+            output_path = Path(self.args.output_path)
```

Comment:
I updated the cli output logic for other clis. Can you take a look?

Reply:
Done.